### DISCUSSION TITLE
How can I test that a failing job triggers my alert

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18178] [Q&A]
Hi all, I'm trying to build out some unit tests for a slack sensor that is set to monitor specific code locations.

One of the tests I want to build out would involve having two different mocked code locations with a job, having that job fail, then ensuring the sensor triggers only in the code location it is monitoring

Is there an example of how to do this anyway in the docs?

The code im looking to test resembles the one in the example <https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors|here>. The specific code im looking at is in the thread below :point_down:

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700520801318879?thread_ts=1700520801.318879&cid=C01U954MEER).*

**Comment by kennycontreras** [comment]
You can look at these examples to set up `slack_on_failure` alerts. It may also be helpful to raise an Exception inside your op and check the behavior of the implemented sensor.

https://docs.dagster.io/_apidocs/libraries/dagster-slack#dagster_slack.slack_on_failure

I hope this helps.

**Reply by CPAPI-104** [reply]
Hi, 

The main part of what I want to do is simulate having jobs in different code locations in the unit test

We have an issue where our slack alert is triggered for a job in multiple code locations/repositories as opposed to only one.

So for example I have 2 jobs:
1 in code location A and one in code location B. I also have a slack on failure alert sensor. I want to ensure that the slack failure sensor only sends a notification for the job in code location/repository B and not A 

Any tips on how to setup a context to test that?

**Comment by benpankow** [comment]
Here are some [internal unit tests](https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster_tests/daemon_sensor_tests/test_run_status_sensors.py#L754) which test out using a run status sensor which monitors across two different code locations - you might be able to pattern your test after this one - running sensor iterations, launching a run, then running a further iteration.

Setting up multiple code locations for a test does require a bit of boilerplate, which you can see in the `instance_with_multiple_code_locations` test utility in the above link.

**Reply by CPAPI-104** [reply]
Thanks this is in line with what I was looking for. Might be worth updating the discussion title for discoverability

Cheers
